.github/workflows/validation-infrastructure-tests.yml

```
@@ -0,0 +1,233 @@
+    if: github.event_name == 'pull_request'
+
+    name: Pre-commit Validation Check
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0  # Need full history for changed files detection
+
+    - name: Set up Python 3.12
+      uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)